Notes.md

```diff
@@ -0,0 +1,50 @@
+##  Questions
+
+1. In CoreDataStorage.swift, what would be a better approax to not have such a convoluted file dealing with ToDos (insert, delete, update...), Categories and future Entities?
+
+
+2. On AddTodoView I have an state that is when the item is saved. On the view at first I wrote this
+```case .saved(let todo):
+    Text(todo.title ?? "")
+        .onAppear { presentationMode.wrappedValue.dismiss() }
+}
+```
+So the idea is when my viewmodel informs me that the item is saved I want to exit the view, but that does not seem particulary right.
+I endup changing that to 
+```
+.onReceive(viewModel.$state, perform: { state in
+    if case .saved(_) = state {
+        presentationMode.wrappedValue.dismiss()
+    }
+})
+```
+What are your thoughts on that? Thinking again about it, both seems right to me, the first one would be nice to show something in the view before dismissing
+```case .saved(let todo):
+    TellTheItemHasCorrectlySaveView()
+        .onAppear { delay(2) { presentationMode.wrappedValue.dismiss() } }
+}
+```
+3. Would you recommend make everything in this protocol asynchronous?
+```
+protocol Storage {
+    func save()
+    func delete(object: Any)
+    func getToDos() -> [ToDo]
+    func addToDo(title: String, dueDate: Date, category: Category) -> ToDo
+    func deleteToDos(indexSet: IndexSet)
+    func getCategories() -> [Category]
+    func addCategory(title: String, imageName: String) -> Category
+}
+```
+Using Coredata with viewContext makes everything continuos, but, if for example I start using a backgroundContext I would have to change the signature, or any other storage framework like Realm
+So would It be better to keep those usually "expensive" task with async signatures? like `func getToDos() -> AnyPublisher<[ToDo]>`
```

Comment:
It really depends on what how you want to implement your abstractions but for the `get` and `add` methods in particular you won't gain anything by making them async. If you want to use the fetched / created objects in your view you'll have to pull them into the viewcontext no matter what, so making it async will not help you in a meaningful way.

If you're considering doing this so you could swap out Core Data later I would still advise to not do it. The number of times I have actually ended up correctly pre-abstracting my code in reality is basically 0. So if you decide to move away from Core Data in the future, you might as well spend a little extra effort at that time because it's typically unlikely to actually make that transition, and it's very likely that you'll need to do some extra work anyway.